colosseum/declaration.py

```diff
@@ -1,29 +1,87 @@
-    ORDER_CHOICES, ORPHANS_CHOICES, OVERFLOW_CHOICES, PADDING_CHOICES,
+    COLOR_CHOICES, DIRECTION_CHOICES, DISPLAY_CHOICES,
+    EMPTY_CELLS_CHOICES, FLEX_BASIS_CHOICES, FLEX_DIRECTION_CHOICES,
+    FLEX_GROW_CHOICES, FLEX_SHRINK_CHOICES, FLEX_START, FLEX_WRAP_CHOICES,
+    FLOAT_CHOICES, GRID_AUTO_CHOICES, GRID_AUTO_FLOW_CHOICES,
+    GRID_GAP_CHOICES, GRID_PLACEMENT_CHOICES, GRID_TEMPLATE_AREA_CHOICES,
+    GRID_TEMPLATE_CHOICES, INITIAL_OUTLINE_VALUES, INLINE, INVERT,
+    JUSTIFY_CONTENT_CHOICES, LETTER_SPACING_CHOICES, LTR, MARGIN_CHOICES,
+    MAX_SIZE_CHOICES, MEDIUM, MIN_SIZE_CHOICES, NORMAL, NOWRAP, ORDER_CHOICES,
+    ORPHANS_CHOICES, OUTLINE_COLOR_CHOICES, OUTLINE_STYLE_CHOICES,
+    OUTLINE_WIDTH_CHOICES, OVERFLOW_CHOICES, PADDING_CHOICES,
     PAGE_BREAK_AFTER_CHOICES, PAGE_BREAK_BEFORE_CHOICES,
     PAGE_BREAK_INSIDE_CHOICES, POSITION_CHOICES, ROW, SEPARATE, SHOW,
     SIZE_CHOICES, STATIC, STRETCH, TABLE_LAYOUT_CHOICES, TEXT_ALIGN_CHOICES,
     TEXT_DECORATION_CHOICES, TEXT_INDENT_CHOICES, TEXT_TRANSFORM_CHOICES, TOP,
     TRANSPARENT, UNICODE_BIDI_CHOICES, VISIBILITY_CHOICES, VISIBLE,
-    WHITE_SPACE_CHOICES, WIDOWS_CHOICES, WORD_SPACING_CHOICES, Z_INDEX_CHOICES,
-    OtherProperty, TextAlignInitialValue, default,
+    WHITE_SPACE_CHOICES, WIDOWS_CHOICES, WORD_SPACING_CHOICES,
+    Z_INDEX_CHOICES, OtherProperty, TextAlignInitialValue, default,
 )
+from .exceptions import ValidationError
+from .wrappers import Outline
 
 _CSS_PROPERTIES = set()
 
 
+def validated_shorthand_property(name, initial, parser, storage_class):
+    """Define the shorthand CSS font property."""
+    if not isinstance(initial, dict):
+        raise ValueError('Initial values for shorthand property must be of type dict!')
+
+    initial = storage_class(**initial)
+
+    def getter(self):
+        shorthand = storage_class()
+        for property_name in initial:
+            try:
+                property_value = getattr(self, '_%s' % property_name)
+                shorthand[property_name] = property_value
+            except AttributeError:
+                pass
+
+        if shorthand:
+            return shorthand
+        else:
+            # It no property has been defined for the outline, return initial values
+            return initial.copy()
+
+    def setter(self, value):
+        try:
+            shorthand = storage_class(**parser(value))
+        except ValidationError:
+            raise ValueError("Invalid value '%s' for CSS property '%s'!" % (value, name))
+
+        # Reset non declared properties to initial values
+        used_properties = shorthand.keys()
+        for property_name, property_value in initial.items():
+            if property_name in used_properties:
+                setattr(self, property_name, shorthand[property_name])
+            else:
+                delattr(self, property_name)
+
+        # We do not explicitely set shorthand properties
+        # setattr(self, '_%s' % name, value)
```

Comment:
We do not actually set the shorthand value, we only set the properties.

Reply:
👍 but we don't need to include the code for what we're not doing :-)

Reply:
Fixed